host-services/builtins/objectstore.go

```diff
@@ -76,16 +76,23 @@ func (o *ObjectStoreService) HandleRequest(
 	} else {
 		nc = conns[hostservices.DefaultConnection]
 	}
+	objectStoreName := metadata[agentapi.BucketContextHeader]
+
+	objectStore, err := o.resolveObjectStore(nc, objectStoreName)
+	if err != nil {
+		o.log.Warn(fmt.Sprintf("failed to resolve object store: %s", err.Error()))
+		return hostservices.ServiceResultFail(500, "unable to resolve object store"), nil
```

Comment:
404?

Reply:
No because the failure to resolve isn't always because it's not there.